### PR TITLE
S4-279: Fix for incorrect IP being stored by patch request

### PR DESCRIPTION
**Pull request by faggc**
- Added recieving IP from patch request

## Description

Added IP to the patch request model so that we can store the correct IP address

## Jira Ticket

https://companieshouse.atlassian.net/browse/S4-279

## Coding Standards

Code Style Guide: https://github.com/companieshouse/dissolution-web/wiki/Code-Style-Guide

## Checklist

- [ ] 3 Amigos
- [x] Acceptance Criteria met
- [x] Branch named {feature|hotfix|task}/{S4-*}
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] All unit tests passing
- [ ] API (Karate) tests passing
- [x] Pulled latest master into feature branch
- [ ] Code reviewed
- [ ] New Docker environment variables have also been added to the revel1 environment and cidev environment
- [ ] If your pull request depends on any other, please link them in the description

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/controller/DissolutionController.java [hunk 1]
```diff
@@ -130,7 +131,7 @@ public DissolutionPatchResponse patchDissolutionApplication(
         }
 
         try {
-            return dissolutionService.addDirectorApproval(companyNumber, userId, request.getRemoteAddr(), body.getOfficerId());
+            return dissolutionService.addDirectorApproval(companyNumber, userId, body.getIpAddress(), body.getOfficerId());
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/controller/DissolutionController.java:134]: can we just pass body through to the service to reduce the number of params? Allow the mappers to extract the fields that they need

File: src/main/java/uk/gov/companieshouse/model/dto/dissolution/DissolutionPatchRequest.java [hunk 2]
```diff
@@ -15,6 +15,9 @@
     @JsonProperty("has_approved")
     private boolean hasApproved;
 
+    @JsonProperty("ip_address")
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/model/dto/dissolution/DissolutionPatchRequest.java:18]: add the NotBlank annotation and add a test case for this in the controller please

Comment (peadarkelly) [reply src/main/java/uk/gov/companieshouse/model/dto/dissolution/DissolutionPatchRequest.java:18]: still missing

Comment (peadarkelly) [reply src/main/java/uk/gov/companieshouse/model/dto/dissolution/DissolutionPatchRequest.java:18]: @faggc This comment is still outstanding

Comment (faggc) [reply src/main/java/uk/gov/companieshouse/model/dto/dissolution/DissolutionPatchRequest.java:18]: Sorry I don't know how that was removed

**Review by peadarkelly — APPROVED** [review]

**Pull request merged**